Build fails in conjunction with ghc-9.6 and latest stackage nightly

Build fails in conjunction with ghc-9.6 and latest stackage nightly:
```
/var/stackage/work/unpack-dir/unpacked/config-ini-0.2.5.0-a2138799480e166f8d7ee42bdfa2812de20508451c5bc39d9228443bec79c879/test/ini-compat/Main.hs:85:30: error: [GHC-39999]
           • Ambiguous type variable ‘f0’ arising from a use of ‘Gen.element’
             prevents the constraint ‘(Foldable f0)’ from being solved.
             Probable fix: use a type annotation to specify what ‘f0’ should be.
             Potentially matching instances:
               instance Foldable (Either a) -- Defined in ‘Data.Foldable’
               instance Foldable Seq.Seq -- Defined in ‘Data.Sequence.Internal’
               ...plus 7 others
               ...plus 47 instances involving out-of-scope types
               (use -fprint-potential-instances to see them all)
           • In the second argument of ‘(<$>)’, namely ‘Gen.element ";#"’
             In the first argument of ‘(<*>)’, namely
               ‘I2.CommentLine <$> Gen.element ";#"’
             In the expression:
               I2.CommentLine <$> Gen.element ";#" <*> textChunk
          |
       85 |           I2.CommentLine <$> Gen.element ";#" <*> textChunk
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -28,6 +28,11 @@ jobs:
     strategy:
       matrix:
         include:
+          - compiler: ghc-9.6.2
+            compilerKind: ghc
+            compilerVersion: 9.6.2
+            setup-method: ghcup
+            allow-failure: false
           - compiler: ghc-9.4.2
             compilerKind: ghc
             compilerVersion: 9.4.2

--- a/test/ini-compat/Main.hs
+++ b/test/ini-compat/Main.hs
@@ -82,7 +82,7 @@ mkComments =
     Gen.list (Range.linear 0 5) $
       Gen.choice
         [ return I2.BlankLine,
-          I2.CommentLine <$> Gen.element ";#" <*> textChunk
+          I2.CommentLine <$> Gen.element (";#" :: String) <*> textChunk
         ]
 
 mkRichIni :: Gen I2.RawIni

